File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
from collections import deque
import math

dq = deque()

def solution(progresses, speeds):
    answer = []
    
    for i in range(len(progresses)) :
        dq.append(math.ceil((100 - progresses[i]) / speeds[i]))
    
    larger = dq.popleft()
    cnt = 1
    
    while (dq) :
        current = dq.popleft()
        if(larger >= current) :
            cnt += 1
        else :
            answer.append(cnt)
            larger = current
            cnt = 1
    
    # 남은 cnt 추가
    answer.append(cnt)
    
    
    # 같은 날짜는 카운트
    # 카운트 리턴
    
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        int[] arr = new int[progresses.length];
        for(int i=0; i<progresses.length; i++) {
            arr[i] = (int)Math.ceil((100.0-progresses[i]) / speeds[i]);
        }
        
        List<Integer> result = new ArrayList<>();
        int mx = arr[0]; int cnt = 1;
        for(int i=1; i<arr.length; i++) {
            if(arr[i] <= mx) {
                cnt++;
            } else {
                mx = arr[i];
                result.add(cnt);
                cnt = 1;
            }
        }
        result.add(cnt);
        
        int[] answer = new int[result.size()];
        for(int i=0; i<result.size(); i++) {
            answer[i] = result.get(i);
        }
        
        return answer;
    }
}